source/sundials/arkode.cc

```diff
@@ -302,33 +294,43 @@ namespace SUNDIALS
         yy        = N_VNew_Serial(system_size);
         abs_tolls = N_VNew_Serial(system_size);
       }
-    reset(data.initial_time, data.initial_step_size, solution);
 
-    double next_time = data.initial_time;
+    DiscreteTime time(data.initial_time,
+                      data.final_time,
+                      data.initial_step_size);
+    reset(time.get_current_time(), time.get_next_step_size(), solution);
 
     if (output_step)
-      output_step(0, solution, 0);
+      output_step(time.get_current_time(), solution, time.get_step_number());
 
-    while (t < data.final_time)
+    while (time.is_at_end() == false)
       {
-        next_time += data.output_period;
+        time.set_desired_next_step_size(data.output_period);
```

Comment:
This means that `data.output_period` is really only a suggested period. Can you document this in the documentation of the class of which `data` is an object?

Reply:
done